### PR TITLE
Prepare OpenApi specification for multiple API versions

### PR DESCRIPTION
**Pull request by daullmer**
**Changes**

- Add OpenApi support for multiple API versions. Each API version generates its own specification file (see URLs below)
- Add demo v2 API controller. I'm not proposing the current place or route as the future place and route. This is simply meant to remind us in the future how we need to implement the new API controllers to show up in the spec files correctly.
- Change test URL according to the new specification location

**Breaking URL changes for OpenApi specification**
The OpenApi specification file moved, so in theory this is a breaking change:
|                   | Current                | New                   | Notes |
|-------------------|------------------------|---------------------------|-------------------------|
| v1 OpenApi specification | /api-docs/openapi.json | /api-docs/v1/openapi.json |
| v2 OpenApi specification |                    | /api-docs/v2/openapi.json |
| Swagger UI       | /api-docs/swagger      | /api-docs/swagger         | Stayed the same. Support for multiple API versions (v1 and v2) |
| ReDoc UI          | /api-docs/redoc        | /api-docs/swagger         | Stayed the same. Doesn't support multiple API versions. Only shows v1 API |

I tried to leave the old specification at its place but that didn't work.

### CODE REVIEW

**Comment by nielsvanvelzen** [comment]
Can we add a "combined" spec? That way we can include both the new and old endpoints in the java apiclient to easily migrate clients.

File: Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -261,18 +277,24 @@ public static IServiceCollection AddJellyfinApiSwagger(this IServiceCollection s
...
-                c.SwaggerDoc("api-docs", new OpenApiInfo
+                var provider = serviceCollection.BuildServiceProvider().GetRequiredService<IApiVersionDescriptionProvider>();
+
+                // Add a swagger document for each discovered API version
+                foreach (var description in provider.ApiVersionDescriptions)
                 {
-                    Title = "Jellyfin API",
-                    Version = "v1",
-                    Extensions = new Dictionary<string, IOpenApiExtension>
+                    c.SwaggerDoc($"{description.GroupName}", new OpenApiInfo()
```
Comment (crobibero) [inline Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs:285]:
```suggestion
                    c.SwaggerDoc(description.GroupName", new OpenApiInfo()
```

File: Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -261,18 +277,24 @@ public static IServiceCollection AddJellyfinApiSwagger(this IServiceCollection s
...
                     {
+                        Title = $"Jellyfin API",
```
Comment (crobibero) [inline Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs:287]:
```suggestion
                        Title = "Jellyfin API",
```

**Comment by Ullmie02** [comment]
I don't think that's possible. We can either have all endpoints in one spec or one spec for each version. I thought we don't want all in one. If we decide to go with one file, we don't need this PR. But I guess it's up to you client guys to decide what we want :P

**Comment by nielsvanvelzen** [comment]
I thought about it and I think that actually having 2 separate documents is better. We can then generate the java apiclient using 2 documents and use a different package/module for both versions. So ignore my previous comment.

File: Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs (hunk 1 above)
Comment (h1dden-da3m0n) [reply Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs:285]: isn't there a obsolete `"` in your code suggestion?

Comment (crobibero) [reply Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs:285]: Thanks, fixed

File: Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -205,6 +207,20 @@ public static IMvcBuilder AddJellyfinApi(this IServiceCollection serviceCollecti
                         options.ForwardLimit = null;
                     }
                 })
+                .AddApiVersioning(
+                o =>
```
Comment (crobibero) [inline Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs:211]:
```suggestion
                .AddApiVersioning(o =>
```

**Review by crobibero — APPROVED** [review]

**Comment by Ullmie02** [comment]
Tried integrating the changes from #5512 but couldn't get it to work. Maybe we can't use this PR if we want to stick to the new versioning

**Pull request closed without merging**